### PR TITLE
Mechanical refactoring of scheduler manager, before changing its behavior

### PR DESCRIPTION
**Pull request by vladsud**
It's mostly mechanical. The only place where I change behavior is testing number | undefined for being falsey - using !== undefined comparison instead.
Class names can be improved - I'll add (including better commenting in future changes)

### CODE REVIEW

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -441,38 +395,110 @@ export class ScheduleManager {
...
+}
+
+/**
+ * This class has the following responsibilities:
+ * 1. It tracks batches as we process ops and raises "batchBegin" and "batchEnd" events.
+ *    As part of it, it validates batch correctness (i.e. no system ops in the middle of batch)
+ * 2. It creates instance of ScheduleManagerCore that ensures we process ops in a batch when we have full batch in.
+ *    It notifies it about
+ */
+export class ScheduleManager {
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/containerRuntime.ts:439]: Can we move ScheduleManager and ScheduleManagerCore into a separate file?

Comment (vladsud) [reply packages/runtime/container-runtime/src/containerRuntime.ts:439]: I'd prefer to do it as separate PR, to allow at least some inspection of code changes (as I try to keep them mechanical for most part, but they are not truly mechanical and I made a mistake in first commit :))

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 2]
```diff
@@ -356,31 +351,134 @@ export class ScheduleManager {
...
+    }
+}
 
-                this.logger.sendTelemetryEvent({
+/**
+ * This class has the following responsibilities:
+ * 1. It tracks batches as we process ops and raises "batchBegin" and "batchEnd" events.
+ *    As part of it, it validates batch correctness (i.e. no system ops in the middle of batch)
+ * 2. It creates instance of ScheduleManagerCore that ensures we process ops in a batch when we have full batch in.
+ *    It notifies it about
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/containerRuntime.ts:449]: nit: Incomplete comment.

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 3]
```diff
@@ -356,31 +351,134 @@ export class ScheduleManager {
...
+            this.setPaused(false);
+        } else {
+            this.setPaused(this.pauseSequenceNumber !== undefined);
+        }
+    }
+}
 
-                this.logger.sendTelemetryEvent({
+/**
+ * This class has the following responsibilities:
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/containerRuntime.ts:445]: Thanks for this explanation! It took me a lot of code inspection to figure this out!

Comment (agarwal-navin) [reply packages/runtime/container-runtime/src/containerRuntime.ts:445]: not sure why this is showing up as a "review comment"

**Comment by agarwal-navin** [comment]
looks good to me. It was a little hard to follow the logic but since it isn't changed, this should be fine.

**Review by agarwal-navin — APPROVED** [review]
